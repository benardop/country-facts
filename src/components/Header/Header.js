import React from 'react';
import { Link } from 'react-router-dom';
import style from './Header.module.css';
import left from '../../assets/images/leftArrow.svg';
import mic from '../../assets/images/mic.svg';
import threeDots from '../../assets/images/threeDots.svg';

function Header() {
  return (

    <div className={style.header}>
      <Link to="/">
        <img src={left} alt="left arrow" />
      </Link>
      <span>Country facts</span>
      <div>
        <img src={mic} alt="microphone" />
        <img src={threeDots} alt="three dots" />
      </div>
    </div>

  );
}

export default Header;
