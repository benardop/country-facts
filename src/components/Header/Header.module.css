@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&family=Puritan&display=swap');

.header {
  width: 100%;
  box-sizing: border-box;
  padding: 2rem;
  font-family: 'lato', sans-serif;
  font-size: 3rem;
  font-weight: 700;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: rgb(88, 137, 230);
  color: white;
}

.header span {
  font-family: 'Puritan', sans-serif;
}

img {
  width: 30px;
  padding: 0.5rem;
}
