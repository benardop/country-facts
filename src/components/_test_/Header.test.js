import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Header from '../Header/Header';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
        ,
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('header-container');
  expect(container).toBeInTheDocument();
});
