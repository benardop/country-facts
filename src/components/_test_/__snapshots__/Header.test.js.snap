// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
Array [
  <div
    className="header"
    data-testid="header-container"
  >
    <a
      data-testid="go-to-home"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="left arrow"
        src="leftArrow.svg"
      />
    </a>
    <span>
      Country facts
    </span>
    <div>
      <img
        alt="microphone"
        src="mic.svg"
      />
      <img
        alt="three dots"
        src="threeDots.svg"
      />
    </div>
  </div>,
  ",",
]
`;
